# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoadRequest.h
# Commit: 98d6cb49f4fd
# Full Hash: 98d6cb49f4fd30dd40b42642879b6509fc3ed12c
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - Create DOM specific Module Loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137298
# ==============================================================================

diff -r e96ec49e3e04 -r 98d6cb49f4fd dom/script/ModuleLoadRequest.h
--- a/dom/script/ModuleLoadRequest.h	Mon Feb 21 15:39:07 2022 +0000
+++ b/dom/script/ModuleLoadRequest.h	Mon Feb 21 15:39:07 2022 +0000
@@ -18,8 +18,7 @@
 namespace dom {
 
 class ModuleScript;
-class ModuleLoader;
-class ScriptLoader;
+class ModuleLoaderBase;
 
 // A reference counted set of URLs we have visited in the process of loading a
 // module graph.
@@ -40,33 +39,18 @@
   ModuleLoadRequest(const ModuleLoadRequest& aOther) = delete;
   ModuleLoadRequest(ModuleLoadRequest&& aOther) = delete;
 
-  ModuleLoadRequest(nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
-                    const SRIMetadata& aIntegrity, nsIURI* aReferrer,
-                    DOMScriptLoadContext* aContext, bool aIsTopLevel,
-                    bool aIsDynamicImport, ModuleLoader* aLoader,
-                    VisitedURLSet* aVisitedSet, ModuleLoadRequest* aRootModule);
-
  public:
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS_INHERITED(ModuleLoadRequest,
                                                          ScriptLoadRequest)
 
-  // Create a top-level module load request.
-  static already_AddRefed<ModuleLoadRequest> CreateTopLevel(
-      nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
-      const SRIMetadata& aIntegrity, nsIURI* aReferrer, ScriptLoader* aLoader,
-      DOMScriptLoadContext* aContext);
+  ModuleLoadRequest(nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
+                    const SRIMetadata& aIntegrity, nsIURI* aReferrer,
+                    DOMScriptLoadContext* aContext, bool aIsTopLevel,
+                    bool aIsDynamicImport, ModuleLoaderBase* aLoader,
+                    VisitedURLSet* aVisitedSet, ModuleLoadRequest* aRootModule);
 
-  // Create a module load request for a static module import.
-  static already_AddRefed<ModuleLoadRequest> CreateStaticImport(
-      nsIURI* aURI, ModuleLoadRequest* aParent);
-
-  // Create a module load request for dynamic module import.
-  static already_AddRefed<ModuleLoadRequest> CreateDynamicImport(
-      nsIURI* aURI, ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-      DOMScriptLoadContext* aContext, ScriptLoader* aLoader,
-      JS::Handle<JS::Value> aReferencingPrivate,
-      JS::Handle<JSString*> aSpecifier, JS::Handle<JSObject*> aPromise);
+  static VisitedURLSet* NewVisitedSetForTopLevelImport(nsIURI* aURI);
 
   bool IsTopLevel() const override { return mIsTopLevel; }
 
@@ -102,7 +86,7 @@
 
   // Pointer to the script loader, used to trigger actions when the module load
   // finishes.
-  RefPtr<ModuleLoader> mLoader;
+  RefPtr<ModuleLoaderBase> mLoader;
 
   // Pointer to the top level module of this module tree, nullptr if this is
   // a top level module